### DISCUSSION TITLE
Are there examples how to use mods with generated code?

### DISCUSSION THREAD

**Discussion by isgj** [stephenafamo/bob#21] [Q&A] [answered]
Hello, I am trying to use the generated code but I cannot make it work with an `or` check. Consider this code

```go
userMod := models.SelectWhere.Users
cs, err := models.Users(ctx, c,
	sm.Where(
		psql.Or(userMod.ID.EQ(users[0].ID),
			userMod.Email.EQ("a@a.com"),
		),
	),
).All()
```

it generates the following query
```
SELECT 
"users"."id" AS "id", ....other columns
FROM "users" AS "users"
WHERE ([("users"."id" = $1)] OR [("users"."email" = $1)])
```
note the extra `[` and `]`

**Comment by stephenafamo** [comment]
This definitely looks like a bug. I'll investigate

**Comment by stephenafamo** [comment, marked as answer]
* `SelectWhere` contains mods which are supposed to be passed directly to the query builder.
* `psql.Or` takes expressions

Your query should be built like this instead:

```go
_, err := models.Users(ctx, db,
	sm.Where(psql.Or(
		models.UserColumns.ID.EQ(psql.Arg(users[0].ID)),
		models.UserColumns.Email.EQ(psql.Arg("a@a.com")),
	)),
).All()
```

Which provides the expected query

```
SELECT
"users"."id" AS "id", ....other columns
FROM "users" AS "users"
WHERE (("users"."id" = $1) OR ("users"."primary_email" = $2))
```

**Reply by isgj** [reply]
Thanks, it worked.

I tried `models.UserColumns.ID.EQ(users[0].ID)` (without psql.Arg) and that didn't quote the argument. Any reason why `.EQ` (and the others) accept `any` and not an expression?

**Reply by stephenafamo** [reply]
I initially assumed that forcing an expression would make simple queries quite tedious to write.

However, it appears that is not the case. In almost all scenarios, the user would want to either use an Arg, a quote or a string literal, all of which are expressions.

This comment actually led to a huge refactor (since everything is quite connected) and now in `v0.18.0`, `EQ` and others now require an Expression. Thank you 🙏🏾 .

**Reply by isgj** [reply]
Hehe, thank you. It's a nice library.
